main.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const core = require('@actions/core');
+const github = require('@actions/github');
+const exec = require('@actions/exec');
+import { chdir } from 'process';
+const glob = require('@actions/glob');
+const artifact = require('@actions/artifact');
+
+
+(async () => {
+  try {
+    const credentials = core.getInput('credentials');
+    const charm_path = core.getInput('charm-path');
+    const charmcraft_channel = core.getInput('charmcraft-channel');
+
+    chdir(charm_path);
+
+    await exec.exec('sudo', [
+      'snap',
+      'install',
+      'charmcraft',
+      '--classic',
+      '--channel',
+      charmcraft_channel,
+    ]);
+
+    await exec.exec('charmcraft', ['pack', '--destructive-mode', '-q']);
```

Comment:
Should destructive mode really be the default?

Reply:
As this will be running on github actions which are isolated environments I think we should default to destructive. The naming is unfortunate, but having to deal with lxd in these kind of environments without any real gains is just not worth it.

Reply:
Agree, CI is a perfectly valid place to use the (badly named) destructive mode

Reply:
Should we remove `-q` to leave some verbosity for when things go wrong?

Reply:
Should we remove `-q` to leave some verbosity for when things go wrong?

Reply:
All that `-q` does is remove progress logging. We also upload the charmcraft logs that should have any relevant information